### PR TITLE
feat: labels (only for assets)

### PR DESCRIPTION
**Pull request by BugGambit**
Need to add integration tests for asset.labels when labels is available in API v1.

### CODE REVIEW

File: src/types/types.ts [hunk 1]
```diff
@@ -543,6 +548,71 @@ export interface DataSetChangeByExternalId extends DataSetPatch, ExternalId {}
...
+   * A list of labels to remove to the resource
+   */
+  remove?: Label[];
+}
+
+export interface ExternalLabelDefinition extends Label {
+  /**
+   * Name of the label.
+   */
+  name: string;
```
Comment (vincent-cognite) [inline src/types/types.ts:585]: no ? makes it required, right?

**Comment by vincent-cognite** [comment]
no support for types in the js sdk? (just curious)

**Review by vincent-cognite — APPROVED** [review]

File: src/types/types.ts (hunk 1 above)
Comment (hhenriques) [reply src/types/types.ts:585]: `name` is required on the API right?

Comment (vincent-cognite) [reply src/types/types.ts:585]: yes (I'm just typescript curious)

File: src/types/types.ts [hunk 2]
```diff
@@ -543,6 +548,71 @@ export interface DataSetChangeByExternalId extends DataSetPatch, ExternalId {}
 
 export interface DataSetChangeById extends DataSetPatch, InternalId {}
 
+export type Label = ExternalId;
```
Comment (polomani) [inline src/types/types.ts:558]: do we need this alias?

File: src/types/types.ts [hunk 3]
```diff
@@ -543,6 +548,71 @@ export interface DataSetChangeByExternalId extends DataSetPatch, ExternalId {}
...
+     * Filter external ids starting with the prefix specified
+     */
+    externalIdPrefix?: ExternalIdPrefix;
+  };
+}
+
+export interface LabelDefinitionFilterRequest
+  extends LabelDefinitionFilter,
+    Cursor,
+    Limit {}
```
Comment (polomani) [inline src/types/types.ts]: instead of `Cursor, Limit`
extends `FilterQuery`

Comment (polomani) [reply src/types/types.ts]: maybe make it similar to `DataSetFilterRequest` so we don't empty types?
I mean move `filter?: LabelDefinitionFilter ` in here and unwrap the current `LabelDefinitionFilter`

**Review by polomani — COMMENTED** [review]
looks good but I am not sure about having `blablaDefinition` everywhere
I would remove alias, use `ExternalId` type and name types as `blavlaLabel` without `Definition`

File: src/types/types.ts (hunk 2 above)
Comment (BugGambit) [reply src/types/types.ts:558]: Yes, for the UX

**Comment by BugGambit** [comment]
> no support for types in the js sdk? (just curious)

@vincent-cognite No, it is not part of v1

**Review by polomani — APPROVED** [review]
don't merge before API release
remember to squash

**Pull request merged**